Meta/CMake/Superbuild/CMakeLists.txt

```diff
@@ -0,0 +1,137 @@
+      elseif(step STREQUAL "build")
+        ExternalProject_Add_StepDependencies("${proj}" build "${proj}-configure")
+      endif()
+    endforeach()
+  endmacro()
+endif()
+
+get_filename_component(
+    SERENITY_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../.."
+    ABSOLUTE CACHE
+)
+set(SERENITY_ARCH "i686" CACHE STRING "Target architecture for SerenityOS.")
+set(SERENITY_TOOLCHAIN "GNU" CACHE STRING "Compliler toolchain to use for Serenity (GNU or Clang)")
+
+# FIXME: It is preferred to keep all the sub-build artifacts below the binary directory for the superbuild
+#        However, this has an impact on developer's IDE settings and more significantly, the Ports tree.
+#        See https://github.com/SerenityOS/serenity/pull/9297#discussion_r697877603
+set(SERENITY_BUILD_DIR_SUFFIX "")
+if(NOT SERENITY_TOOLCHAIN STREQUAL "GNU")
+  string(TOLOWER "${SERENITY_TOOLCHAIN}" SERENITY_BUILD_DIR_SUFFIX)
+endif()
+set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}${SERENITY_BUILD_DIR_SUFFIX}")
+
+if (CMAKE_HOST_SYSTEM_NAME MATCHES "SerenityOS")
+    message(STATUS "Good job on building cmake!")
+else()
+    configure_file("${SERENITY_SOURCE_DIR}/Toolchain/CMake/${SERENITY_TOOLCHAIN}Toolchain.txt.in" "${SERENITY_BUILD_DIR}/CMakeToolchain.txt" @ONLY)
+    set(SERENITY_TOOLCHAIN_FILE "${SERENITY_BUILD_DIR}/CMakeToolchain.txt" CACHE PATH "Toolchain file to use for cross-compilation")
+    # Support non-cross builds by stuffing this in a variable
+    set(SERENITY_TOOLCHAIN_FILE_ARG "-DCMAKE_TOOLCHAIN_FILE:STRING=${SERENITY_TOOLCHAIN_FILE}")
+endif()
+
+# Allow the Ninja generators to output messages as they happen by assigning
+# these jobs to the 'console' job pool
+set(console_access "")
+if(CMAKE_GENERATOR MATCHES "^Ninja")
+  set(
+      console_access
+      USES_TERMINAL_CONFIGURE YES
+      USES_TERMINAL_BUILD YES
+      USES_TERMINAL_INSTALL YES
+  )
+endif()
+
+include(ExternalProject)
+
+# Collect options for Lagom build
+set(lagom_options "")
+macro(serenity_option name)
+    set(${ARGV})
+    list(APPEND lagom_options "-D${name}:STRING=${${name}}")
+endmacro()
+include("${SERENITY_SOURCE_DIR}/Meta/CMake/lagom_options.cmake")
+
+# Forward user defined host toolchain to lagom build
+if (DEFINED CMAKE_C_COMPILER)
+    set(CMAKE_C_COMPILER "${CMAKE_C_COMPILER}" CACHE STRING "C Compiler to use for host builds")
+    list(APPEND lagom_options "-DCMAKE_C_COMPILER:STRING=${CMAKE_C_COMPILER}")
+endif()
+if (DEFINED CMAKE_CXX_COMPILER)
+    set(CMAKE_CXX_COMPILER "${CMAKE_CXX_COMPILER}" CACHE STRING "C++ Compiler to use for host builds")
+    list(APPEND lagom_options "-DCMAKE_CXX_COMPILER:STRING=${CMAKE_CXX_COMPILER}")
+endif()
+
+ExternalProject_Add(
+    lagom
+    SOURCE_DIR "${SERENITY_SOURCE_DIR}/Meta/Lagom"
+    BINARY_DIR "${PROJECT_BINARY_DIR}/../lagom"
+    INSTALL_DIR "${PROJECT_BINARY_DIR}/../lagom-install"
+    EXCLUDE_FROM_ALL YES
+    CMAKE_CACHE_DEFAULT_ARGS
```

Comment:
Now that I'm looking, you are passing the cache options using the `DEFAULT` variant here, which means that after the first configure with the superbuild, the values will not be propagated, because these are just defaults. The `CMAKE_CACHE_ARGS` argument should be used, which will use the `FORCE` option when setting the variables. I think all of them can be passed via `CMAKE_CACHE_ARGS`, because they are either cache variables in the superbuild or are dependent on code in this lists file.